COLLABORATOR_GUIDE.md

```diff
@@ -802,7 +802,7 @@ pulled from the staging branch into the LTS branch only when a release is
 being prepared and may be pulled into the LTS branch in a different order
 than they were landed in staging.
 
-Any Collaborator may land commits into a staging branch, but only the release
+Any Collaborator may land commits into a non-LTS staging branch. Only the release
```

Comment:
While we're clarifying, should we clarify what is meant by release team? For example, I am in the nodejs/release team because I am in nodejs/citgm but I am not in the nodejs/releasers nor nodejs/backporters nor nodejs/lts teams.